packages/core/src/internal-components/overlay/overlay.element.ts

```diff
@@ -14,12 +16,26 @@ import {
   internalProperty,
   onKey,
   property,
+  AnimationModalEnterName,
+  reverseAnimation,
 } from '@cds/core/internal';
 import { html, query } from 'lit-element';
 import { styles } from './overlay.element.css.js';
 
-export function isNestedOverlay(myId: string, overlayPrefix: string, trapIds: string[]): boolean {
+export function isNestedOverlay(
+  myId: string,
+  overlayPrefix: string,
+  trapIds: string[],
+  previousValue?: boolean
+): boolean {
   const overlayIds = trapIds.filter(id => id.indexOf(overlayPrefix) > -1);
+
+  if (previousValue === true && trapIds.indexOf(myId) < 0) {
+    // handling situation where focusTrapIds remove our overlay from the list and we still need to consider
+    // this overlay as nested. this happens when an overlay is being closed/hidden
```

Comment:
This is the bad state that animations revealed. It could've happened with any a synchronous update of a focus trap. Animations just revealed it first.